src/TableFunctions/TableFunctionDictionary.cpp

```diff
@@ -42,10 +46,33 @@ void TableFunctionDictionary::parseArguments(const ASTPtr & ast_function, Contex
 ColumnsDescription TableFunctionDictionary::getActualTableStructure(ContextPtr context) const
 {
     const ExternalDictionariesLoader & external_loader = context->getExternalDictionariesLoader();
+    std::string resolved_name = external_loader.resolveDictionaryName(dictionary_name, context->getCurrentDatabase());
+    auto load_result = external_loader.load(resolved_name);
```

Comment:
Previously, `external_loader.getDictionaryStructure()` (l. 45, left side) did not load the dictionary, now we do load the dictionary. Intended?

Reply:
yeah, It's easy to know the type of Dictionary after loading it.